public/app/features/dashboard-scene/utils/interactions.ts

```diff
@@ -0,0 +1,127 @@
+    reportDashboardInteraction('sharing_public_email_revoke_clicked', { ...properties });
+  },
+  resendPublicDashboardEmailClicked: (properties?: Record<string, unknown>) => {
+    reportDashboardInteraction('sharing_public_email_resend_clicked', { ...properties });
+  },
+  publicDashboardEmailInviteClicked: (properties?: Record<string, unknown>) => {
+    reportDashboardInteraction('sharing_public_email_invite_clicked', { ...properties });
+  },
+  publicDashboardShareTypeChange: (properties?: Record<string, unknown>) => {
+    reportDashboardInteraction('sharing_public_can_view_clicked', { ...properties });
+  },
+  publicDashboardTimeSelectionChanged: (properties?: Record<string, unknown>) => {
+    reportDashboardInteraction('sharing_public_time_picker_clicked', { ...properties });
+  },
+  publicDashboardAnnotationsSelectionChanged: (properties?: Record<string, unknown>) => {
+    reportDashboardInteraction('sharing_public_annotations_clicked', { ...properties });
+  },
+  publicDashboardUrlCopied: (properties?: Record<string, unknown>) => {
+    reportDashboardInteraction('sharing_public_copy_url_clicked', { ...properties });
+  },
+  publicDashboardPauseSharingClicked: (properties?: Record<string, unknown>) => {
+    reportDashboardInteraction('sharing_public_pause_clicked', { ...properties });
+  },
+  revokePublicDashboardClicked: (properties?: Record<string, unknown>) => {
+    reportDashboardInteraction('sharing_public_revoke_clicked', { ...properties });
+  },
+
+  // Empty dashboard state interactions:
+  emptyDashboardButtonClicked: (properties?: Record<string, unknown>) => {
+    reportDashboardInteraction('emptydashboard_clicked', { ...properties });
+  },
+
+  // Toolbar interactions
+  toolbarAddButtonClicked: (properties?: Record<string, unknown>) => {
+    reportDashboardInteraction('toolbar_add_clicked', { ...properties });
+  },
+  toolbarFavoritesClick: () => {
+    reportDashboardInteraction('toolbar_actions_clicked', { item: 'favorites' });
+  },
+  toolbarSettingsClick: () => {
+    reportDashboardInteraction('toolbar_actions_clicked', { item: 'settings' });
+  },
+  toolbarRefreshClick: () => {
+    reportDashboardInteraction('toolbar_actions_clicked', { item: 'refresh' });
+  },
+  toolbarTimePickerClick: () => {
+    reportDashboardInteraction('toolbar_actions_clicked', { item: 'time_picker' });
+  },
+  toolbarZoomClick: () => {
+    reportDashboardInteraction('toolbar_actions_clicked', { item: 'zoom_out_time_range' });
+  },
+
+  toolbarShareClick: () => {
+    reportDashboardInteraction('toolbar_actions_clicked', { item: 'share' });
+  },
+  toolbarSaveClick: () => {
+    reportDashboardInteraction('toolbar_actions_clicked', { item: 'save' });
+  },
+
+  toolbarAddClick: () => {
+    reportDashboardInteraction('toolbar_actions_clicked', { item: 'add' });
+  },
+};
+
+const reportDashboardInteraction: typeof reportInteraction = (name, ...args) => {
```

Comment:
To not check the 1 by 1 events, are the names of any events changing? To make sure we mark them as deprecated and migrate the corresponding dashboards. I guess they are not, but just in case